### PR TITLE
Added comments for connection message sending

### PR DESCRIPTION
**Pull request by bentoi**
This PR adds comments to the C# connection code to better document how sending of a message works.

### CODE REVIEW

File: csharp/src/Ice/ConnectionI.cs [hunk 1]
```diff
@@ -1902,29 +1902,44 @@ private bool validate(int operation)
...
     }
 
+
+    /// <summary>
+    /// Sends the next queued messages. This is called by message() when the message from _writeStream was fully sent so
+    /// before sending the next message the message which was being sent (_sendStreams.First) is first removed from the
+    /// queue and if it has a sent callback to call, the sent callback is queued.
+    /// </summary>
+    /// <param name="callbacks">The sent callbacks to call for the messages that were sent.</param>
+    /// <returns></returns>
```
Comment (bernardnormier) [inline csharp/src/Ice/ConnectionI.cs:1912]: Missing doc for return value.

File: csharp/src/Ice/ConnectionI.cs [hunk 2]
```diff
@@ -1902,29 +1902,44 @@ private bool validate(int operation)
         return true;
     }
 
+
+    /// <summary>
+    /// Sends the next queued messages. This is called by message() when the message from _writeStream was fully sent so
```
Comment (bernardnormier) [inline csharp/src/Ice/ConnectionI.cs:1907]: I would break the second sentence in several sentences, at least add commas.

**Review by bernardnormier — APPROVED** [review]

File: csharp/src/Ice/ConnectionI.cs [hunk 3]
```diff
@@ -1902,29 +1902,44 @@ private bool validate(int operation)
...
+
         try
         {
             while (true)
             {
                 //
-                // Notify the message that it was sent.
+                // The message that was being sent is sent so we can swap back the write stream buffer to the
+                // outgoing message (required for retry) and queue its sent callback if there's a sent callback to
+                // call.
```
Comment (bernardnormier) [inline csharp/src/Ice/ConnectionI.cs:1942]: After line 1945, is _writeStream empty? If not, what is in _writeStream?

Comment (bentoi) [reply csharp/src/Ice/ConnectionI.cs]: Yes, `_writeStream` is empty. I've added documentation to the `_readStream` and `_writeStream` data members.

**Pull request merged**